### ISSUE TITLE
[Scope] Require ID.me Identify proofing for all DS Logon accounts for DD access

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#2844]
Related to #1798

### Problem

DS Logon's 2FA isn't very secure, and it's caused people's accounts to be compromised. We do not want this to affect people on VA.gov, especially if it could compromise their DD info.

### Solution

If a user is logged in with DS Logon but has not done ID.me identity proofing, put the user through ID.me identify proofing flow before allowing them to access direct deposit.

### Additional details

In a meeting on 10.29.19, Erik, Anna, and I considered gating VA.gov for all DS Logon users and requiring they all go through ID.me's verification. The rationale is that this might be worthwhile if we have significant concerns about DS Logon as a whole.

We know that hundreds of thousands of people are DS Logon LOA3 and we do not want to inconvenience hundreds of thousands of people by asking them to verify their identities again. However, we can not currently tell whether these people have verified via DS Logon or ID.me. Anna is currently working to figure this out. If most people had already verified through ID.me, we may be able to move forward gating all other DS Logon users. However, if most are verified through DS Logon and not ID.me, we will gate only the direct deposit feature as to not inconvenience a significant amount of VA.gov users.

### Definition of done

- [ ] Scope solution for this (we'll create a separate ticket for design)

**Comment by Samara-Strauss** [comment]
Update: Anna still working to assess how many DS Logon LOA3 users are verified through DS Logon vs. ID.me. Hoping to have a PR out (https://github.com/department-of-veterans-affairs/vets-api/pull/3498) to help us gather this info by 11/1. This means we might have some helpful stats by Monday 11/4.

**Comment by Justin-Pickett** [comment]
Need to follow up with @annaswims to gather the info needed for Samara to finish designs.

**Comment by annaswims** [comment]
I think these are the metrics you're looking for 
http://grafana.vfs.va.gov/d/000000033/site-authentication?orgId=1&fullscreen&panelId=24

and a more detailed description
https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Products/Identity/Login/IDme/201810_IDme_Whitelabeled_Flow/%5B%2313806%5D%20Ensure%20that%20the%20whitelabled%20ID.me%20identity%20proofing%20flow%20appears%20for%20DS%20Logon%20and%20MHV%20users.md#overview

**Comment by Samara-Strauss** [comment]
@annaswims — so this is saying that the majority of DS Logon users (~245K) are verified through DS logon and not ID.me, correct?

If so, then we should just gate direct deposit and not the whole site for DS Logon users to cause the least amount of disruption to folks. 

@erikphansen and @annaswims — I am having trouble remembering the flow we discussed for how this would work. Can you help me remember? I believe we talked about this:

- Person who is DS Logon LOA3 logs in to VA.gov and goes to the profile and sees a prompt in the direct deposit section to verify their identity again with ID.me.
- They go to verify their identity through ID.me.
- This is the part I don't quite remember — the ideal experience is that they come back to the profile and can access direct deposit on all subsequent visits, but I recall us talking about how the flow might be more complicated than this because of when we check to see if they've verified through ID.me. Will they need to enter a code for 2FA for the direct deposit section each time they want to access it? This sounds familiar to me but I'm not sure if I am remembering things incorrectly. If it's easier, let's hop on a quick call to talk it through.

**Comment by Samara-Strauss** [comment]
OK, after having talked through the craziness that is log on and a bunch of super complex paths we could take, we decided on this: require that DS logon users sign in with ID.me or MHV to access the direct deposit feature. 

We chose this approach for three reasons:

1. The other solutions we talked about that had to do with detecting ID.me verification for DS Logon users were super complicated and would have messed with some of our core login set up. Not only would this be a poor use of developer time and add a lot of complexity, it puts us at risk for messing up something that has to do with our core login functionality. I don't think that risk is worth it for this use case.
2. Gating this feature alone is better than gating the whole site — DS logon users will still have access to 99% of the VA.gov functionality. 
3. Updating direct deposit is not something people need to do often, so while this will be an inconvenience for DS Logon users, it'll be a rare one.
